zbackup_base.hh

```diff
@@ -34,8 +34,8 @@ class ZBackupBase: public Paths
 public:
   DEF_EX( Ex, "ZBackup exception", std::exception )
   DEF_EX_STR( exWontOverwrite, "Won't overwrite existing file", Ex )
-  DEF_EX( exStdinError, "Error reading from standard input", Ex )
-  DEF_EX( exWontReadFromTerminal, "Won't read data from a terminal", exStdinError )
+  DEF_EX_STR( exInputError, "Error reading from input:", Ex )
+  DEF_EX( exWontReadFromTerminal, "Won't read data from a terminal", Ex )
```

Comment:
Are you talking about the third parameter of this line?  

     DEF_EX( exWontReadFromTerminal, "Won't read data from a terminal", Ex )

Assuming that is the case...  Because exStdinError is no longer appropriate and has been genericized to exInputError, and changed to use DEF_EX_STR so that the file path can be included in the error message.  Deriving exWontReadFromTerminal from exInputError results in a compilation failure, as there is nothing to provide its ctor argument:

<pre>
zbackup_base.hh: In constructor ‘ZBackupBase::exWontReadFromTerminal::exWontReadFromTerminal()’:
zbackup_base.hh:38:3: error: no matching function for call to ‘ZBackupBase::exInputError::exInputError()’
zbackup_base.hh:38:3: note: candidates are:
zbackup_base.hh:37:3: note: ZBackupBase::exInputError::exInputError(const char*, unsigned int)
zbackup_base.hh:37:3: note:   candidate expects 2 arguments, 0 provided
zbackup_base.hh:37:3: note: ZBackupBase::exInputError::exInputError(const string&)
zbackup_base.hh:37:3: note:   candidate expects 1 argument, 0 provided
zbackup_base.hh:37:3: note: ZBackupBase::exInputError::exInputError(const ZBackupBase::exInputError&)
zbackup_base.hh:37:3: note:   candidate expects 1 argument, 0 provided
zutils.cc: In member function ‘void ZBackup::backupFromStdin(const string&)’:
zutils.cc:25:34: note: synthesized method ‘ZBackupBase::exWontReadFromTerminal::exWontReadFromTerminal()’ first required here 
</pre>

If I have misunderstood your comment, please clarify or suggest a better alternative.